ast/ast.hpp

```diff
@@ -233,20 +281,70 @@ struct BinaryExpression : Expression {
     void accept(Visitor& v) override { v.visit(this); }
 };
 
-struct Primary : Expression {
-    lexer::Token value;
+struct Primary : Expression, Statement {
     bool operator==(const Primary& other) const {
-        return Node::operator==(other) && value == other.value;
+        return Node::operator==(other);
+    }
+    virtual void accept(Visitor& v) override { v.visit(this); }
+};
+
+struct IntegerLiteral : Primary {
```

Comment:
Right. It was because the grammar specifies that any of these can act as a primary, but I forgot that we are using the terminology a little bit differently. Only `Identifier` should be a primary because it can later be replaced with a routine call